### DISCUSSION TITLE
SKonsole: Streamlining PR and Commit Message Authoring with Semantic Kernel

### DISCUSSION THREAD

**Discussion by lemillermicrosoft** [microsoft/semantic-kernel#106] [Show and tell]
Are you tired of spending too much time crafting the perfect commit message or PR feedback? **SKonsole** might just be the tool you need. This console app, built on Semantic Kernel, uses various skills to generate commit messages and pull request feedback.

The **SKonsole** repository contains projects and classes that help automate the process of authoring pull requests and commit messages. The `PRSkill`, for example, can generate feedback, commit messages, and pull request descriptions based on `git diff` or `git show` output. This skill is built on top of the `CondenseSkill`, which is a skill that can condense multiple chunks of text into a single chunk.

The app requires the Semantic Kernel as a dependency and can be found on the GitHub repository: [https://github.com/lemillermicrosoft/skonsole](https://github.com/lemillermicrosoft/skonsole). While the app currently supports only two skills, the repository includes instructions on how to expand the app's capabilities.

**Note** that this is just the beginning of my exploration of the Semantic Kernel's capabilities. In the future, I plan to add more features such as bot chat, planning, semantic memory, code generation, and more. Stay tuned for updates!

**Comment by wonhyeongseo** [comment]
I love the idea and will try using it in my side project. Thank you for your awesome code, @lemillermicrosoft !

**Comment by geirolsen** [comment]
Great idea Lee. Along similar lines I would love to have an AI code reviewer that I can do a pre-review with before I send a broader code review to my crew. It would handle coding conventions, compliance related policies, SAL annotations etc. Have you put some thought into this type of feature set? This is especially powerful on larger organizations and code bases where conventions can differ across the source tree, policies are updated on a cadence etc.

**Reply by lemillermicrosoft** [reply]
These are great ideas! I've already considered and started work on a pre-review feature (it's in a branch for now) -- lots of room for making this better, too. Thinking how it could be powerful to index a repository PR history with commits + feedback into semantic Memory and use that to help guide a review process to give quality feedback and guidance.

Compliance and more would be even more powerful!

**Comment by lemillermicrosoft** [comment]
I've updated the tool and made it available for installation via 

```cmd
dotnet tool install --global SKonsole
```

**Reply by lemillermicrosoft** [reply]
A new version is available today, v1.0.1 -- thanks to @xbotter for their continued support in making the tool more usable and feature packed. 

```cmd
dotnet tool update --global SKonsole
```
